src/Raven.Server/Documents/Handlers/Processors/Attachments/AttachmentHandlerProcessorForGetAttachment.cs

```diff
@@ -1,111 +1,86 @@
-                    if (HttpContext.Response.Headers.ContainsKey(Constants.Headers.ContentType))
-                        HttpContext.Response.Headers.Remove(Constants.Headers.ContentType);
-                }
-
-                HttpContext.Response.Headers[Constants.Headers.AttachmentHash] = attachment.Base64Hash.ToString();
-                HttpContext.Response.Headers[Constants.Headers.AttachmentSize] = attachment.Size.ToString();
-                HttpContext.Response.Headers[Constants.Headers.Etag] = $"\"{attachment.ChangeVector}\"";
+            if (attachment == null)
+            {
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                return;
+            }
 
-                var (sendBody, start, bytesRemaining) = RangeHelper.SetRangeHeaders(HttpContext, attachment.Size);
-                if (!sendBody)
-                    return;
+            IGetAttachmentStrategy strategy = attachment.RetireParameters.IsRetiredAttachment()
+                  ? new RetiredGetAttachmentStrategyProcessor(RequestHandler)
+                  : new RegularGetAttachmentStrategyProcessor(RequestHandler);
 
-                using (context.GetMemoryBuffer(out var buffer))
-                await using (var stream = attachment.Stream)
-                {
-                    if (start > 0)
-                    {
-                        stream.Seek(start, SeekOrigin.Begin);
-                    }
+            strategy.CheckAttachmentFlagAndConfigurationAndThrowIfNeeded(context, attachment, documentId, name);
 
-                    var responseStream = RequestHandler.ResponseBodyStream();
-                    while (true)
-                    {
-                        if (bytesRemaining is <= 0)
-                        {
-                            return;
-                        }
+            var attachmentChangeVector = RequestHandler.GetStringFromHeaders(Constants.Headers.IfNoneMatch);
+            if (attachmentChangeVector == attachment.ChangeVector)
+            {
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+                return;
+            }
 
-                        var readLength = buffer.Size;
-                        if (bytesRemaining.HasValue)
-                        {
-                            readLength = (int)Math.Min(bytesRemaining.Value, readLength);
-                        }
+            try
+            {
+                var fileName = Path.GetFileName(attachment.Name);
+                fileName = Uri.EscapeDataString(fileName);
+                HttpContext.Response.Headers[Constants.Headers.ContentDisposition] = $"attachment; filename=\"{fileName}\"; filename*=UTF-8''{fileName}";
+            }
+            catch (ArgumentException e)
+            {
+                if (Logger.IsDebugEnabled)
+                    Logger.Debug($"Skip Content-Disposition header because of not valid file name: {attachment.Name}", e);
+            }
```

Comment:
Why is this just a log? What would be the output in such a scenario?
What kind of a filename can trigger this?